libcheckout

```diff
@@ -1,5 +1,5 @@
 function checkout() {
-  git clone $SEMAPHORE_GIT_URL $SEMAPHORE_GIT_DIR
+  git clone --branch $SEMAPHORE_GIT_BRANCH --single-branch $SEMAPHORE_GIT_URL $SEMAPHORE_GIT_DIR
```

Comment:
Let's use the one that we have in Semaphore 1. This is tricky area. https://github.com/renderedtext/job_runner/blob/48fcb67ea1dc786e77198c50a572ba3964f6086d/lib/command_queue/repository/clone.rb#L32

Reply:
@darkofabijan isn't almost the same, that one used `-b` instead of `--branch`.

@bmarkons what is the benefit of adding `--single-branch`?

Reply:
I understand this can break every Job run.

The `--branch` option is identical to `-b`, while `--single-branch` tells git to only clone the history of a single branch specified with `--branch`.

I have used these options together before, but for the sake of confidence here is the example output:
```
✲ dev git clone --branch bm/finisher-rabbit-event --single-branch git@github.com:renderedtext/zebra.git ~/zebra
Cloning into '/home/vagrant/zebra'...
remote: Counting objects: 553, done.
remote: Compressing objects: 100% (325/325), done.
remote: Total 553 (delta 246), reused 474 (delta 172), pack-reused 0
Receiving objects: 100% (553/553), 181.14 KiB | 0 bytes/s, done.
Resolving deltas: 100% (246/246), done.
Checking connectivity... done.

✲ dev cd ~/zebra

✲ zebra (bm/finisher-rabbit-event) git status
On branch bm/finisher-rabbit-event
Your branch is up-to-date with 'origin/bm/finisher-rabbit-event'.

nothing to commit, working directory clean

✲ zebra (bm/finisher-rabbit-event) git branch
* bm/finisher-rabbit-event
```

Reply:
Thanks for the clarification. I vote for removing the `--single-branch` based on the previous comment.

Two reasons:

- It will make a bigger impact if someone wants to cache their git repository (if they choose to)
- It allows jumping from one branch to another and behaves like on your local machine, someone might need this for some complex task (for example comparing diffs betweean master and feature branch, and running only affected tests)

Reply:
👌 Removing `--single-branch` option.